streams/src/main/java/org/apache/kafka/streams/state/internals/WrappedStateStore.java

```diff
@@ -19,81 +19,66 @@
-     * @return the state store this store directly wraps
-     */
-    StateStore wrappedStore();
-
-    abstract class AbstractStateStore implements WrappedStateStore {
-        final StateStore innerState;
-
-        protected AbstractStateStore(final StateStore inner) {
-            this.innerState = inner;
+public class WrappedStateStore<S extends StateStore> implements StateStore {
+    public static boolean isTimestamped(StateStore stateStore) {
+        if (stateStore instanceof TimestampedBytesStore) {
+            return true;
+        } else if (stateStore instanceof WrappedStateStore) {
+            return isTimestamped(((WrappedStateStore) stateStore).wrappedStore());
+        } else {
+            return false;
         }
+    }
 
-        @Override
-        public void init(final ProcessorContext context,
-                         final StateStore root) {
-            innerState.init(context, root);
-        }
+    private final S innerState;
 
-        @Override
-        public String name() {
-            return innerState.name();
-        }
+    public WrappedStateStore(final S innerState) {
+        this.innerState = innerState;
+    }
 
-        @Override
-        public boolean persistent() {
-            return innerState.persistent();
-        }
+    @Override
+    public void init(final ProcessorContext context,
+                     final StateStore root) {
+        innerState.init(context, root);
+    }
 
-        @Override
-        public boolean isOpen() {
-            return innerState.isOpen();
-        }
+    @Override
+    public String name() {
+        return innerState.name();
+    }
 
-        void validateStoreOpen() {
-            if (!innerState.isOpen()) {
-                throw new InvalidStateStoreException("Store " + innerState.name() + " is currently closed.");
-            }
-        }
+    @Override
+    public boolean persistent() {
+        return innerState.persistent();
+    }
 
-        @Override
-        public StateStore inner() {
-            if (innerState instanceof WrappedStateStore) {
-                return ((WrappedStateStore) innerState).inner();
-            }
-            return innerState;
-        }
+    @Override
+    public boolean isOpen() {
+        return innerState.isOpen();
+    }
 
-        @Override
-        public void flush() {
-            innerState.flush();
+    public void validateStoreOpen() {
+        if (!innerState.isOpen()) {
+            throw new InvalidStateStoreException("Store " + innerState.name() + " is currently closed.");
         }
+    }
 
-        @Override
-        public void close() {
-            innerState.close();
-        }
+    @Override
+    public void flush() {
+        innerState.flush();
+    }
 
-        @Override
-        public StateStore wrappedStore() {
-            return innerState;
-        }
+    @Override
+    public void close() {
+        innerState.close();
+    }
 
+    public S wrappedStore() {
+        return innerState;
```

Comment:
`wrappedStore()` should `return wrapped;` -- that's why I argue for renaming the variable.